### PR TITLE
CONSOLE-4266: Fix CSP inline script tag error in Console HTML template

### PR DESCRIPTION
**Pull request by vojtechszocs**
This PR addresses a [CSP](https://developer.mozilla.org/en-US/docs/Web/HTTP/CSP) violation due to [Console HTML index template](https://github.com/openshift/console/blob/master/frontend/public/index.html) containing an inline `<script>` tag.

By default, CSP standard treats inline `<script>` tags as [unsafe](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src#unsafe_inline_script).

This particular `<script>` tag is part of the Console application, so we allow it within our policy via the [nonce](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#nonce) attribute.

> A cryptographic nonce (number used once) to allow scripts in a [script-src Content-Security-Policy](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/script-src). The server must generate a unique nonce value each time it transmits a policy. It is critical to provide a nonce that cannot be guessed as bypassing a resource's policy is otherwise trivial.

This PR also does a bit of refactoring around `crypto/rand` related functions, removing code duplication.

### 1. Basic verification steps

1. `git clone` Console repo, switch to latest `master` branch, build Console backend and frontend, `oc login` to cluster and run the Bridge server (enabling [dynamic demo plugin](https://github.com/openshift/console/tree/master/dynamic-demo-plugin) is **_not_** necessary). Visit `http://localhost:9000/` in your browser, the above CSP error :point_up:  should appear in browser dev tools "Console" tab.
1. Apply changes from this PR, rebuild Console backend (rebuilding frontend is **_not_** necessary) and rerun the Bridge server. Open Console in your web browser, the CSP error as mentioned above should not appear.

### 2. CSP standard compliance verification steps

1. Inspect HTML and check the presence of `nonce` attribute on the inline `<script>` tag.

2. Inspect network activity and check the presence of `'nonce-{value}'` within the CSP response header.

3. Reload the page and repeat steps 1. and 2. above. **The `nonce` value should be different upon each reload.**

cc @jhadvig @spadgett

### CODE REVIEW

**Comment by vojtechszocs** [comment]
/hold cancel

Local testing shows that everything works as expected.

**Review by TheRealJon — APPROVED** [review]
/lgtm

File: pkg/server/server.go
Comment (jhadvig) [inline pkg/server/server.go:274]: not new, but could you please update the error handling similarly as you did below?

File: pkg/utils/utils.go
Comment (jhadvig) [inline pkg/utils/utils.go:25]:
```suggestion
	if err != nil {
		return "", err
	}
	return encoding.EncodeToString(randomBytes)[:length], nil
```

**Review by jhadvig — COMMENTED** [review]
Two comments, otherwise the change looks good 👍

File: pkg/server/server.go
Comment (vojtechszocs) [reply pkg/server/server.go:274]: As discussed, we can do this as a follow-up.

In general there are two different cases, errors from which we can recover vs. error from which we cannot recover and therefore we exit the process.

**Comment by vojtechszocs** [comment]
PR updated - addressed the review comment in `RandomString` function.

**Review by jhadvig — APPROVED** [review]
/lgtm

**Comment by jhadvig** [comment]
QE Approver:
/assign @yapei
Docs Approver:
/assign @opayne1
PX Approver:
/assign @reestr

**Comment by yapei** [comment]
Tested the PR and no issues found
/label qe-approved

**Pull request merged**
